src/main/java/com/neu/wham/services/GetEventServiceImpl.java

```diff
@@ -57,6 +58,8 @@
 		builder.addParameter("location.longitude", lon);
 		builder.addParameter("location.within", radius + "mi");
 		builder.addParameter("token", "DXVHSQKC2T2GGBTUPOY2");
+		if(q != null)

```

Comment:
is (null != q), a better way? if q is null and we are trying to check against a value, we might run into null pointer issues :)